docs/design_docs/proxy.md

```diff
@@ -0,0 +1,511 @@
+
+#### 6.4 Interaction with Query Coordinator
+
+For LoadCollection, LoadPartition, ReleaseCollection, ReleasePartition requests, the Proxy directly forwards these
+requests to Query Coordinator for execution after checking and preprocessing these requests. When the Proxy receives
+feedback from Query Coordiantor, it returns the feedback results to the clients.
+
+The semantics of the Load operation is to load Collection or Partition from persistent storage into the memory of Query
+Nodes, or import streaming data into QueryNode so that it can be queried. If the load operation is not performed, the
+query operation on the Collection or Partition cannot be performed. For the Load operation, Query Coordinator is
+responsible for allocating DmChannels to different queryNodes and subscribing to them, and is responsible for receiving
+those stream data. QueryCoordinator also allocates and loads the segments that have been persisted in storage in
+Query Nodes.
+
+The semantics of the Release operation is the reverse operation of the Load operation, and the function is to unload the
+data of the Collection or Partition from the memory. For Release operations, Query Coordinator is responsible for
+notifying query nodes to unload the corresponding Collection or Partition in memory, and then sending the
+ReleaseDqlMessageStream command to Root Coordinator, and Root Coordinator is responsible for broadcasting the
+ReleaseDqlMessageStream command to all Proxies, so that all related stream used to send search request and receive
+search result in Proxy will be closed.
+
+The other interaction between Proxy and Query Coordinator is that Proxy needs to query Query Coordinator for statistics
+about Collection, Partition, and Segment. Taking ShowCollections as an example, if the ShowCollections parameter
+specifies that the query is for Collections that have been loaded into memory, the ShowCollection request will be
+forwarded to QueryCoordinator, and QueryCoordinator will return a list of all the recorded Collections loaded into
+memory. Taking LoadCollection as another example, itss synchronization semantics is that the number of rows loaded in
+the memory must be no less than the number of rows that have been persisted. This requires the Proxy to ask the Query
+Coordinator for the sum of the number of rows currently loaded into the query nodes in the Collection.
+
+#### 6.5 Decouple Functionality and Communication
+
+![decouple](./graphs/decouple.jpeg)
+
+As shown in the figure above, there are interactions between various types of components in the Milvus2.0 system. The
+implementation of these interactions will vary according to the deployment of Milvus. Milvus Standalone allows the
+various components of Milvus as a whole independent process to be deployed on a single node. Milvus Cluster distributes
+all components on multiple nodes. In Milvus Standalone, the interaction between components can be parameter transfer
```

Comment:
The answer is yes. According to the our design of deploying standalone milvus, the interaction between different components should be function call between these components. We use a `Client` interface to encapsulate these communications, but now the `Client` interface only have a grpc implementation, we don't distinguish the deployment in `Client`, this may be our goal in later versions.